--- exec-tool.in.old	2018-12-07 14:15:12.564917669 -0800
+++ exec-tool.in	2018-12-07 15:42:00.674978944 -0800
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/bash
 
 # Copyright (C) 2007-2017 Free Software Foundation, Inc.
 # This file is part of GCC.
@@ -73,6 +73,14 @@
     ;;
 esac
 
+args=()
+for i in "${1+"$@"}"; do
+        if echo "$i" | grep -q "^[A-Za-z]:"; then
+                args+=("`winepath --unix \"$i\"`")
+        else
+                args+=("$i")
+        fi
+done
 case "$original" in
   ../*)
     # compute absolute path of the location of this script
@@ -80,29 +88,29 @@
     scriptdir=`cd "$tdir" && pwd`
 
     if test -x $scriptdir/../$dir/$prog; then
-      test "$fast_install" = yes || exec $scriptdir/../$dir/$prog ${1+"$@"}
+      test "$fast_install" = yes || exec $scriptdir/../$dir/$prog ${args[@]}
 
       # if libtool did everything it needs to do, there's a fast path
       lt_prog=$scriptdir/../$dir/$objdir/lt-$prog 
-      test -x $lt_prog && exec $lt_prog ${1+"$@"}
+      test -x $lt_prog && exec $lt_prog ${args[@]}
 
       # libtool has not relinked ld-new yet, but we cannot just use the
       # previous stage (because then the relinking would just never happen!).
       # So we take extra care to use prev-ld/ld-new *on recursive calls*.
       eval LT_RCU="\${LT_RCU_$id}"
-      test x"$LT_RCU" = x"1" && exec $scriptdir/../prev-$dir/$prog ${1+"$@"}
+      test x"$LT_RCU" = x"1" && exec $scriptdir/../prev-$dir/$prog ${args[@]}
 
       eval LT_RCU_$id=1
       export LT_RCU_$id
-      $scriptdir/../$dir/$prog ${1+"$@"}
+      $scriptdir/../$dir/$prog ${args[@]}
       result=$?
       exit $result
 
     else
-      exec $scriptdir/../prev-$dir/$prog ${1+"$@"}
+      exec $scriptdir/../prev-$dir/$prog ${args[@]}
     fi
     ;;
   *)
-    exec $original ${1+"$@"}
+    exec $original ${args[@]}
     ;;
 esac
