--- config-ml.in.old	2020-03-12 09:34:54.930114000 -0700
+++ config-ml.in	2020-03-12 09:35:44.010408528 -0700
@@ -215,7 +215,7 @@
 if [ "${ml_toplevel_p}" = yes ]; then
 
 multidirs=
-for i in `${CC-gcc} --print-multi-lib 2>/dev/null`; do
+for i in `${CC-gcc} --print-multi-lib 2>/dev/null | sed -e 's/\r//g'`; do
   dir=`echo $i | sed -e 's/;.*$//'`
   if [ "${dir}" = "." ]; then
     true
@@ -496,7 +496,7 @@
 	  srcrootpre=`cd $(srcdir); ${PWD_COMMAND}`/; export srcrootpre; \
 	  lib=`echo "$${rootpre}" | sed -e 's,^.*/\([^/][^/]*\)/$$,\1,'`; \
 	  compiler="$(CC)"; \
-	  for i in `$${compiler} --print-multi-lib 2>/dev/null`; do \
+	  for i in `$${compiler} --print-multi-lib 2>/dev/null | sed -e 's/\r//g'`; do \
 	    dir=`echo $$i | sed -e 's/;.*$$//'`; \
 	    if [ "$${dir}" = "." ]; then \
 	      true; \
@@ -739,7 +739,7 @@
     fi
 
     # find compiler flag corresponding to ${ml_dir}
-    for i in `${CC-gcc} --print-multi-lib 2>/dev/null`; do
+    for i in `${CC-gcc} --print-multi-lib 2>/dev/null | sed -e 's/\r//g'`; do
       dir=`echo $i | sed -e 's/;.*$//'`
       if [ "${dir}" = "${ml_dir}" ]; then
         flags=`echo $i | sed -e 's/^[^;]*;//' -e 's/@/ -/g'`
