--- configure.old	2019-05-26 22:03:59.343579801 -0700
+++ configure	2019-05-26 22:13:24.145225508 -0700
@@ -685,13 +685,19 @@
 libgcc_visibility
 gcc_cv_readelf
 gcc_cv_objdump
+WINE_ORIGINAL_NM_FOR_TARGET
 ORIGINAL_NM_FOR_TARGET
 gcc_cv_nm
+WINE_ORIGINAL_LD_GOLD_FOR_TARGET
+WINE_ORIGINAL_LD_BFD_FOR_TARGET
 ORIGINAL_LD_GOLD_FOR_TARGET
 ORIGINAL_LD_BFD_FOR_TARGET
+WINE_ORIGINAL_LD_FOR_TARGET
 ORIGINAL_LD_FOR_TARGET
+WINE_ORIGINAL_PLUGIN_LD_FOR_TARGET
 ORIGINAL_PLUGIN_LD_FOR_TARGET
 gcc_cv_ld
+WINE_ORIGINAL_AS_FOR_TARGET
 ORIGINAL_AS_FOR_TARGET
 gcc_cv_as
 enable_fast_install
@@ -879,7 +885,8 @@
 PATH_SEPARATOR
 SHELL'
 ac_subst_files='option_includes
-language_hooks'
+language_hooks
+wine_find_str_workarround'
 ac_user_opts='
 enable_option_checking
 with_build_libsubdir
@@ -18448,7 +18455,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 18451 "configure"
+#line 18458 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -18554,7 +18561,7 @@
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<_LT_EOF
-#line 18557 "configure"
+#line 18564 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -21882,10 +21889,13 @@
 
 case "$ORIGINAL_AS_FOR_TARGET" in
   ./as | ./as$build_exeext) ;;
-  *) ac_config_files="$ac_config_files as:exec-tool.in"
+  *) ac_config_files="$ac_config_files as.cmd:exec-tool.cmd.in"
  ;;
 esac
 
+WINE_ORIGINAL_AS_FOR_TARGET=`winepath -w $gcc_cv_as`
+
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking what assembler to use" >&5
 $as_echo_n "checking what assembler to use... " >&6; }
 if test "$gcc_cv_as" = ../gas/as-new$build_exeext; then
@@ -22042,6 +22052,7 @@
 fi
 
 ORIGINAL_PLUGIN_LD_FOR_TARGET=$gcc_cv_ld
+WINE_ORIGINAL_PLUGIN_LD_FOR_TARGET=`winepath -w $gcc_cv_ld`
 PLUGIN_LD_SUFFIX=`basename $gcc_cv_ld | sed -e "s,$target_alias-,,"`
 # if the PLUGIN_LD is set ld-new, just have it as ld
 # as that is the installed named.
@@ -22054,12 +22065,14 @@
 if test "${with_plugin_ld+set}" = set; then :
   withval=$with_plugin_ld; if test x"$withval" != x; then
    ORIGINAL_PLUGIN_LD_FOR_TARGET="$withval"
+   WINE_ORIGINAL_PLUGIN_LD_FOR_TARGET=`winepath -w "$withval"`
    PLUGIN_LD_SUFFIX="$withval"
  fi
 fi
 
 
 
+
 cat >>confdefs.h <<_ACEOF
 #define PLUGIN_LD_SUFFIX "$PLUGIN_LD_SUFFIX"
 _ACEOF
@@ -22121,10 +22134,13 @@
 
 case "$ORIGINAL_LD_FOR_TARGET" in
   ./collect-ld | ./collect-ld$build_exeext) ;;
-  *) ac_config_files="$ac_config_files collect-ld:exec-tool.in"
+  *) ac_config_files="$ac_config_files collect-ld.cmd:exec-tool.cmd.in"
  ;;
 esac
 
+WINE_ORIGINAL_LD_FOR_TARGET=`winepath -w $gcc_cv_ld`
+
+
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking what linker to use" >&5
 $as_echo_n "checking what linker to use... " >&6; }
 if test "$gcc_cv_ld" = ../ld/ld-new$build_exeext \
@@ -22158,6 +22174,8 @@
 	gcc_cv_gld_minor_version=`expr "$gcc_cv_gld_version" : "VERSION=[0-9]*\.\([0-9]*\)"`
 	ORIGINAL_LD_BFD_FOR_TARGET=../ld/ld-new$build_exeext
 	ORIGINAL_LD_GOLD_FOR_TARGET=../gold/ld-new$build_exeext
+	WINE_ORIGINAL_LD_BFD_FOR_TARGET=..\\ld\\ld-new$build_exeext
+	WINE_ORIGINAL_LD_GOLD_FOR_TARGET=..\\gold\\ld-new$build_exeext
 else
 	{ $as_echo "$as_me:${as_lineno-$LINENO}: result: $gcc_cv_ld" >&5
 $as_echo "$gcc_cv_ld" >&6; }
@@ -22165,11 +22183,15 @@
 	gcc_cvs_ld_program=`dirname $gcc_cv_ld`/`basename $gcc_cv_ld $host_exeext`
 	ORIGINAL_LD_BFD_FOR_TARGET=${gcc_cvs_ld_program}.bfd$host_exeext
 	ORIGINAL_LD_GOLD_FOR_TARGET=${gcc_cvs_ld_program}.gold$host_exeext
+	WINE_ORIGINAL_LD_BFD_FOR_TARGET=`winepath -w ${gcc_cvs_ld_program}.bfd$host_exeext`
+	WINE_ORIGINAL_LD_GOLD_FOR_TARGET=`winepath -w ${gcc_cvs_ld_program}.gold$host_exeext`
 fi
 
 
 
 
+
+
 # Figure out what nm we will be using.
 gcc_cv_binutils_srcdir=`echo $srcdir | sed -e 's,/gcc$,,'`/binutils
 if test "${gcc_cv_nm+set}" = set; then :
@@ -22245,9 +22267,11 @@
 
 case "$ORIGINAL_NM_FOR_TARGET" in
   ./nm | ./nm$build_exeext) ;;
-  *) ac_config_files="$ac_config_files nm:exec-tool.in"
+  *) ac_config_files="$ac_config_files nm.cmd:exec-tool.cmd.in"
  ;;
 esac
+WINE_ORIGINAL_NM_FOR_TARGET=`winepath -w $gcc_cv_nm`
+
 
 
 # Figure out what objdump we will be using.
@@ -30211,6 +30235,13 @@
 # Configure the subdirectories
 # AC_CONFIG_SUBDIRS($subdirs)
 
+${LN_S} $srcdir/wineFindStrWorkarround.exe wineFindStrWorkarround.exe
+wine_find_str_workarround="wineFindStrWorkarround.exe"
+
+
+ac_config_files="$ac_config_files exec-tool.c:exec-tool.c.in"
+
+
 # Create the Makefile
 # and configure language subdirectories
 ac_config_files="$ac_config_files $all_outputs"
@@ -30916,10 +30947,11 @@
     "auto-host.h") CONFIG_HEADERS="$CONFIG_HEADERS auto-host.h:config.in" ;;
     "depdir") CONFIG_COMMANDS="$CONFIG_COMMANDS depdir" ;;
     "gccdepdir") CONFIG_COMMANDS="$CONFIG_COMMANDS gccdepdir" ;;
-    "as") CONFIG_FILES="$CONFIG_FILES as:exec-tool.in" ;;
-    "collect-ld") CONFIG_FILES="$CONFIG_FILES collect-ld:exec-tool.in" ;;
-    "nm") CONFIG_FILES="$CONFIG_FILES nm:exec-tool.in" ;;
+    "as.cmd") CONFIG_FILES="$CONFIG_FILES as.cmd:exec-tool.cmd.in" ;;
+    "collect-ld.cmd") CONFIG_FILES="$CONFIG_FILES collect-ld.cmd:exec-tool.cmd.in" ;;
+    "nm.cmd") CONFIG_FILES="$CONFIG_FILES nm.cmd:exec-tool.cmd.in" ;;
     "clearcap.map") CONFIG_LINKS="$CONFIG_LINKS clearcap.map:${srcdir}/config/$clearcap_map" ;;
+    "exec-tool.c") CONFIG_FILES="$CONFIG_FILES exec-tool.c:exec-tool.c.in" ;;
     "$all_outputs") CONFIG_FILES="$CONFIG_FILES $all_outputs" ;;
     "default") CONFIG_COMMANDS="$CONFIG_COMMANDS default" ;;
 
@@ -31539,9 +31571,9 @@
   do
       ${CONFIG_SHELL-/bin/sh} $ac_aux_dir/mkinstalldirs $lang/$DEPDIR
   done ;;
-    "as":F) chmod +x as ;;
-    "collect-ld":F) chmod +x collect-ld ;;
-    "nm":F) chmod +x nm ;;
+    "as.cmd":F) chmod +x as.cmd ;;
+    "collect-ld.cmd":F) chmod +x collect-ld.cmd ;;
+    "nm.cmd":F) chmod +x nm.cmd ;;
     "default":C)
 case ${CONFIG_HEADERS} in
   *auto-host.h:config.in*)
