--- Makefile.in.old	2022-03-03 22:11:39.931794578 -0800
+++ Makefile.in	2022-03-14 15:53:22.595610460 -0700
@@ -1945,7 +1945,7 @@
 # "nul" instead of "/dev/null" if we're building on a mingw machine.
 # Specify the path to gcc/testsuite/selftests within the srcdir
 # as an argument to -fself-test.
-DEVNULL=$(if $(findstring mingw,$(build)),nul,/dev/null)
+DEVNULL=nul
 SELFTEST_FLAGS = -nostdinc $(DEVNULL) -S -o $(DEVNULL) \
 	-fself-test=$(srcdir)/testsuite/selftests
 
@@ -2011,6 +2011,7 @@
 # Dump a specs file to make -B./ read these specs over installed ones.
 $(SPECS): xgcc$(exeext)
 	$(GCC_FOR_TARGET) -dumpspecs > tmp-specs
+	dos2unix tmp-specs
 	mv tmp-specs $(SPECS)
 
 # We do want to create an executable named `xgcc', so we can use it to
@@ -3015,6 +3016,7 @@
 macro_list: s-macro_list; @true
 s-macro_list : $(GCC_PASSES) cc1$(exeext)
 	echo | $(GCC_FOR_TARGET) -E -dM - | \
+	  sed -e 's/\r//g' | \
 	  sed -n -e 's/^#define \([^_][a-zA-Z0-9_]*\).*/\1/p' \
 		 -e 's/^#define \(_[^_A-Z][a-zA-Z0-9_]*\).*/\1/p' | \
 	  sort -u > tmp-macro_list
@@ -3026,10 +3028,10 @@
 # Build up a list of multilib directories and corresponding sysroot
 # suffixes, in form sysroot;multilib.
 	if $(GCC_FOR_TARGET) -print-sysroot-headers-suffix > /dev/null 2>&1; then \
-	  set -e; for ml in `$(GCC_FOR_TARGET) -print-multi-lib`; do \
+	  set -e; for ml in `$(GCC_FOR_TARGET) -print-multi-lib | sed -e 's/\r//g'`; do \
 	    multi_dir=`echo $${ml} | sed -e 's/;.*$$//'`; \
 	    flags=`echo $${ml} | sed -e 's/^[^;]*;//' -e 's/@/ -/g'`; \
-	    sfx=`$(GCC_FOR_TARGET) $${flags} -print-sysroot-headers-suffix`; \
+	    sfx=`$(GCC_FOR_TARGET) $${flags} -print-sysroot-headers-suffix | sed -e 's/\r//g'`; \
 	    if [ "$${multi_dir}" = "." ]; \
 	      then multi_dir=""; \
 	    else \
