--- configure.ac.old	2019-05-26 21:50:00.123411031 -0700
+++ configure.ac	2019-05-26 22:13:14.021275328 -0700
@@ -2176,9 +2176,12 @@
 AC_SUBST(ORIGINAL_AS_FOR_TARGET)
 case "$ORIGINAL_AS_FOR_TARGET" in
   ./as | ./as$build_exeext) ;;
-  *) AC_CONFIG_FILES(as:exec-tool.in, [chmod +x as]) ;;
+  *) AC_CONFIG_FILES(as.cmd:exec-tool.cmd.in, [chmod +x as.cmd]) ;;
 esac 
 
+WINE_ORIGINAL_AS_FOR_TARGET=`winepath -w $gcc_cv_as`
+AC_SUBST(WINE_ORIGINAL_AS_FOR_TARGET)
+
 AC_MSG_CHECKING(what assembler to use)
 if test "$gcc_cv_as" = ../gas/as-new$build_exeext; then
   # Single tree build which includes gas.  We want to prefer it
@@ -2262,6 +2265,7 @@
 fi])
 
 ORIGINAL_PLUGIN_LD_FOR_TARGET=$gcc_cv_ld
+WINE_ORIGINAL_PLUGIN_LD_FOR_TARGET=`winepath -w $gcc_cv_ld`
 PLUGIN_LD_SUFFIX=`basename $gcc_cv_ld | sed -e "s,$target_alias-,,"`
 # if the PLUGIN_LD is set ld-new, just have it as ld
 # as that is the installed named.
@@ -2273,9 +2277,11 @@
 [AS_HELP_STRING([[--with-plugin-ld=[ARG]]], [specify the plugin linker])],
 [if test x"$withval" != x; then
    ORIGINAL_PLUGIN_LD_FOR_TARGET="$withval"
+   WINE_ORIGINAL_PLUGIN_LD_FOR_TARGET=`winepath -w "$withval"`
    PLUGIN_LD_SUFFIX="$withval"
  fi])
 AC_SUBST(ORIGINAL_PLUGIN_LD_FOR_TARGET)
+AC_SUBST(WINE_ORIGINAL_PLUGIN_LD_FOR_TARGET)
 AC_DEFINE_UNQUOTED(PLUGIN_LD_SUFFIX, "$PLUGIN_LD_SUFFIX", [Specify plugin linker])
 
 # Check to see if we are using gold instead of ld
@@ -2329,9 +2335,12 @@
 AC_SUBST(ORIGINAL_LD_FOR_TARGET)
 case "$ORIGINAL_LD_FOR_TARGET" in
   ./collect-ld | ./collect-ld$build_exeext) ;;
-  *) AC_CONFIG_FILES(collect-ld:exec-tool.in, [chmod +x collect-ld]) ;;
+  *) AC_CONFIG_FILES(collect-ld.cmd:exec-tool.cmd.in, [chmod +x collect-ld.cmd]) ;;
 esac 
 
+WINE_ORIGINAL_LD_FOR_TARGET=`winepath -w $gcc_cv_ld`
+AC_SUBST(WINE_ORIGINAL_LD_FOR_TARGET)
+
 AC_MSG_CHECKING(what linker to use)
 if test "$gcc_cv_ld" = ../ld/ld-new$build_exeext \
    || test "$gcc_cv_ld" = ../gold/ld-new$build_exeext; then
@@ -2367,16 +2376,22 @@
 changequote([,])dnl
 	ORIGINAL_LD_BFD_FOR_TARGET=../ld/ld-new$build_exeext
 	ORIGINAL_LD_GOLD_FOR_TARGET=../gold/ld-new$build_exeext
+	WINE_ORIGINAL_LD_BFD_FOR_TARGET=..\\ld\\ld-new$build_exeext
+	WINE_ORIGINAL_LD_GOLD_FOR_TARGET=..\\gold\\ld-new$build_exeext
 else
 	AC_MSG_RESULT($gcc_cv_ld)
 	in_tree_ld=no
 	gcc_cvs_ld_program=`dirname $gcc_cv_ld`/`basename $gcc_cv_ld $host_exeext`
 	ORIGINAL_LD_BFD_FOR_TARGET=${gcc_cvs_ld_program}.bfd$host_exeext
 	ORIGINAL_LD_GOLD_FOR_TARGET=${gcc_cvs_ld_program}.gold$host_exeext
+	WINE_ORIGINAL_LD_BFD_FOR_TARGET=`winepath -w ${gcc_cvs_ld_program}.bfd$host_exeext`
+	WINE_ORIGINAL_LD_GOLD_FOR_TARGET=`winepath -w ${gcc_cvs_ld_program}.gold$host_exeext`
 fi
 
 AC_SUBST(ORIGINAL_LD_BFD_FOR_TARGET)
 AC_SUBST(ORIGINAL_LD_GOLD_FOR_TARGET)
+AC_SUBST(WINE_ORIGINAL_LD_BFD_FOR_TARGET)
+AC_SUBST(WINE_ORIGINAL_LD_GOLD_FOR_TARGET)
 
 # Figure out what nm we will be using.
 gcc_cv_binutils_srcdir=`echo $srcdir | sed -e 's,/gcc$,,'`/binutils
@@ -2407,8 +2422,10 @@
 AC_SUBST(ORIGINAL_NM_FOR_TARGET)
 case "$ORIGINAL_NM_FOR_TARGET" in
   ./nm | ./nm$build_exeext) ;;
-  *) AC_CONFIG_FILES(nm:exec-tool.in, [chmod +x nm]) ;;
+  *) AC_CONFIG_FILES(nm.cmd:exec-tool.cmd.in, [chmod +x nm.cmd]) ;;
 esac
+WINE_ORIGINAL_NM_FOR_TARGET=`winepath -w $gcc_cv_nm`
+AC_SUBST(WINE_ORIGINAL_NM_FOR_TARGET)
 
 
 # Figure out what objdump we will be using.
@@ -6590,6 +6607,12 @@
 # Configure the subdirectories
 # AC_CONFIG_SUBDIRS($subdirs)
 
+${LN_S} $srcdir/wineFindStrWorkarround.exe wineFindStrWorkarround.exe
+wine_find_str_workarround="wineFindStrWorkarround.exe"
+AC_SUBST_FILE(wine_find_str_workarround)
+
+AC_CONFIG_FILES(exec-tool.c:exec-tool.c.in)
+
 # Create the Makefile
 # and configure language subdirectories
 AC_CONFIG_FILES($all_outputs)
