--- fixed-obj.mk.old	2019-06-02 20:37:13.023550891 -0700
+++ fixed-obj.mk	2019-06-02 20:39:03.754722966 -0700
@@ -24,10 +24,20 @@
 
 ifneq ($o,$(filter $o,$(LIB2FUNCS_EXCLUDE)))
 $o$(objext): %$(objext): $(srcdir)/fixed-bit.c
-	$(gcc_compile) -DL$($*-label) $($*-opt) -c $(srcdir)/fixed-bit.c $(vis_hide)
+	$(gcc_compile) -DL$($*-label) $($*-opt) -c $(srcdir)/fixed-bit.c $(vis_hide) && cp $(dep_filename) $(dep_filename).bak && (cat $(dep_filename) \
+	  | egrep '^\s*[A-Za-z]:' | sed -e 's/^\s*//g' -e 's/:\s*$$//g' -e 's/\s*\\\s*$$//g' | sort -u | sed -e 's/\\/\\\\/g' | \
+	    while IFS= read -r line; \
+	    do \
+	      sed -ie 's#'"$$line"'#'$$(winepath -u "$$line" | sed -e 's#:#\\\\:#g')'#g' $(dep_filename).bak; \
+	    done) && mv $(dep_filename).bak $(dep_filename)
 
 ifeq ($(enable_shared),yes)
 $(o)_s$(objext): %_s$(objext): $(srcdir)/fixed-bit.c
-	$(gcc_s_compile) -DL$($*-label) $($*-opt) -c $(srcdir)/fixed-bit.c
+	$(gcc_s_compile) -DL$($*-label) $($*-opt) -c $(srcdir)/fixed-bit.c && cp $(dep_filename) $(dep_filename).bak && (cat $(dep_filename) \
+	  | egrep '^\s*[A-Za-z]:' | sed -e 's/^\s*//g' -e 's/:\s*$$//g' -e 's/\s*\\\s*$$//g' | sort -u | sed -e 's/\\/\\\\/g' | \
+	    while IFS= read -r line; \
+	    do \
+	      sed -ie 's#'"$$line"'#'$$(winepath -u "$$line" | sed -e 's#:#\\\\:#g')'#g' $(dep_filename).bak; \
+	    done) && mv $(dep_filename).bak $(dep_filename)
 endif
 endif
