--- interfaces-and-libraries.sh.old	2022-04-10 11:21:27.930117864 -0700
+++ interfaces-and-libraries.sh	2022-04-12 23:22:49.210276509 -0700
@@ -58,7 +58,7 @@
     echo "  - a directory with C header files (usually \"CIncludes\")"
     echo "  - a directory with Rez header files (usually \"RIncludes\")"
     echo "  - (for 68K) a directory containing Interface.o (usually \"Libraries\")"
-    echo "  - (for PPC) a directory containing InterfaceLib (usually \"SharedLibraries\")"
+    echo "  - (for PPC) a directory containing InterfaceLib.bin (usually \"SharedLibraries\")"
     echo "  - (for Carbon) Carbon.h and CarbonLib, in the same directories"
     echo
     echo "The Interfaces&Libraries folder from Apple's last MPW release (MPW 3.5 "
@@ -116,11 +116,11 @@
 
     if [ $BUILD_PPC != false ]; then
 
-        if locateInterfaceThing INTERFACELIB InterfaceLib; then
+        if locateInterfaceThing INTERFACELIB InterfaceLib.bin; then
             SHAREDLIBRARIES=`dirname "$INTERFACELIB"`
         else
             SHAREDLIBRARIES=""
-            echo "Could not find InterfaceLib, using included import libraries."
+            echo "Could not find InterfaceLib.bin, using included import libraries."
         fi
 
         if locateInterfaceThing OPENTRANSPORTAPPPPC OpenTransportAppPPC.o; then
@@ -151,7 +151,7 @@
         if locateInterfaceThing CARBONLIB CarbonLib; then
             carbondir=`dirname "$CARBONLIB"`
             if [ "$carbondir" != "$SHAREDLIBRARIES" ]; then
-                echo "CarbonLib found, but not in the same directory as InterfaceLib."
+                echo "CarbonLib found, but not in the same directory as InterfaceLib.bin."
                 echo "This is confusing."
                 echo
                 explainInterfaces
@@ -248,9 +248,9 @@
 
             rm -f $lib
 
-            if ConvertObj "$macobj" > "$asm"; then
-                m68k-apple-macos-as "$asm" -o "$obj"
-                m68k-apple-macos-ar cqs "$lib" "$obj"
+	    if $PREFIX/bin/ConvertObj.exe "$macobj" > "$asm"; then
+                $PREFIX/bin/m68k-apple-macos-as.exe "$asm" -o "$obj"
+                $PREFIX/bin/m68k-apple-macos-ar.exe cqs "$lib" "$obj"
             fi
             rm -f "$asm"
         fi
@@ -261,7 +261,7 @@
 {
     DEST=${1:-"$PREFIX/universal"}
     mkdir -p "$DEST/libppc"
-    case `ResInfo -n "$INTERFACELIB" 2> /dev/null || echo 0` in
+    case `($PREFIX/bin/ResInfo.exe -n "$INTERFACELIB" 2> /dev/null | sed -e 's/\r//g') || echo 0` in
         0)
             if [ -n "$INTERFACELIB" ]; then
                 echo "WARNING: Couldn't read resource fork for \"$INTERFACELIB\"."
@@ -276,7 +276,7 @@
                 libname=`basename "$shlib"`
                 implib=lib${libname%.bin}.a
                 printf "    %30s => %-30s\n" ${libname} ${implib}
-                MakeImport "$shlib" "$DEST/libppc/$implib" || true
+                $PREFIX/bin/MakeImport.exe "$shlib" "$DEST/libppc/$implib" || true
             done
             ;;
     esac
@@ -291,7 +291,7 @@
                 # and wrap it in a .a archive for convenience
                 lib="$DEST"/libppc/lib$basename.a
                 rm -f "$lib"
-                powerpc-apple-macos-ar cqs "$lib" "$obj"
+                $PREFIX/bin/powerpc-apple-macos-ar.exe cqs "$lib" "$obj"
             fi
         done
     fi
