diff -Naur Retro68.old/libelf/CMakeLists.txt Retro68/libelf/CMakeLists.txt
--- Retro68.old/libelf/CMakeLists.txt	2022-03-02 21:58:34.567147334 -0800
+++ Retro68/libelf/CMakeLists.txt	2022-03-19 13:27:38.980208476 -0700
@@ -1,7 +1,9 @@
-file(GLOB ELF_FILES src/*.c src/*.h include/*.h)
+file(GLOB ELF_FILES src/*.cpp src/*.c src/*.h include/*.h)
+file(GLOB SPECIAL_ELF_FILES src/fchmod.cpp)
 add_library(ELF ${ELF_FILES}
 	)
 target_include_directories(ELF PRIVATE src)
 target_include_directories(ELF PUBLIC include)
 target_link_libraries(ELF -lz)
 target_compile_definitions(ELF PRIVATE HAVE_CONFIG_H)
+set_property(SOURCE ${SPECIAL_ELF_FILES} PROPERTY COMPILE_FLAGS -mgeneral-regs-only)
diff -Naur Retro68.old/libelf/include/libelf.h Retro68/libelf/include/libelf.h
--- Retro68.old/libelf/include/libelf.h	2022-03-02 21:58:34.643141982 -0800
+++ Retro68/libelf/include/libelf.h	2022-03-02 21:53:52.515465039 -0800
@@ -187,6 +187,8 @@
   ELF_K_NUM
 } Elf_Kind;
 
+#define uid_t int
+#define gid_t int
 
 /* Archive member header.  */
 typedef struct
diff -Naur Retro68.old/libelf/src/common.h Retro68/libelf/src/common.h
--- Retro68.old/libelf/src/common.h	2022-03-02 21:58:34.647141701 -0800
+++ Retro68/libelf/src/common.h	2022-03-02 22:02:55.084553964 -0800
@@ -30,7 +30,29 @@
 #ifndef _COMMON_H
 #define _COMMON_H       1
 
-#include <ar.h>
+//#include <ar.h>
+#ifndef INCLUDED_AR_H
+#define INCLUDED_AR_H
+#define ARMAG  "!<arch>\n"
+#define SARMAG  8
+
+#define ARFMAG  "`\n"
+
+struct ar_hdr
+{
+    char ar_name[ 16 ];
+    char ar_date[ 12 ];
+    char ar_uid[ 6 ];
+    char ar_gid[ 6 ];
+    char ar_mode[ 8 ];
+    char ar_size[ 10 ];
+    char ar_fmag[ 2 ];
+};
+
+#define SARFMAG  2
+#define SARHDR  sizeof( struct ar_hdr )
+#endif
+
 #include <stdlib.h>
 #include <string.h>
 #include "portable_endian.h"
diff -Naur Retro68.old/libelf/src/elf32_updatefile.c Retro68/libelf/src/elf32_updatefile.c
--- Retro68.old/libelf/src/elf32_updatefile.c	2022-03-02 21:58:34.647141701 -0800
+++ Retro68/libelf/src/elf32_updatefile.c	2022-03-02 22:03:34.263874612 -0800
@@ -38,7 +38,9 @@
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <sys/mman.h>
+//#include <sys/mman.h>
+#include "mman.h"
+#include "sysconf.h"
 
 #include <system.h>
 #include "libelfP.h"
diff -Naur Retro68.old/libelf/src/elf_begin.c Retro68/libelf/src/elf_begin.c
--- Retro68.old/libelf/src/elf_begin.c	2022-03-02 21:58:34.647141701 -0800
+++ Retro68/libelf/src/elf_begin.c	2022-03-02 22:04:00.931498487 -0800
@@ -39,7 +39,9 @@
 #include <stddef.h>
 #include <string.h>
 #include <unistd.h>
-#include <sys/mman.h>
+//#include <sys/mman.h>
+#include "mman.h"
+#include "fcntl.h"
 #include <sys/stat.h>
 
 #include <system.h>
diff -Naur Retro68.old/libelf/src/elf_end.c Retro68/libelf/src/elf_end.c
--- Retro68.old/libelf/src/elf_end.c	2022-03-02 21:58:34.647141701 -0800
+++ Retro68/libelf/src/elf_end.c	2022-03-02 22:04:28.163178250 -0800
@@ -34,7 +34,8 @@
 #include <assert.h>
 #include <stddef.h>
 #include <stdlib.h>
-#include <sys/mman.h>
+//#include <sys/mman.h>
+#include "mman.h"
 
 #include "libelfP.h"
 
diff -Naur Retro68.old/libelf/src/elf_update.c Retro68/libelf/src/elf_update.c
--- Retro68.old/libelf/src/elf_update.c	2022-03-02 21:58:34.651141419 -0800
+++ Retro68/libelf/src/elf_update.c	2022-03-02 22:06:06.734463257 -0800
@@ -34,7 +34,10 @@
 #include <libelf.h>
 #include <fcntl.h>
 #include <unistd.h>
-#include <sys/mman.h>
+//#include <sys/mman.h>
+#include "mman.h"
+#include "fchmod.h"
+#include "fallocate.h"
 #include <sys/stat.h>
 
 #include "libelfP.h"
diff -Naur Retro68.old/libelf/src/libelfP.h Retro68/libelf/src/libelfP.h
--- Retro68.old/libelf/src/libelfP.h	2022-03-02 21:58:34.655141137 -0800
+++ Retro68/libelf/src/libelfP.h	2022-03-02 22:15:59.232446714 -0800
@@ -30,7 +30,30 @@
 #ifndef _LIBELFP_H
 #define _LIBELFP_H 1
 
-#include <ar.h>
+//#include <ar.h>
+
+#ifndef INCLUDED_AR_H
+#define INCLUDED_AR_H
+#define ARMAG  "!<arch>\n"
+#define SARMAG  8
+
+#define ARFMAG  "`\n"
+
+struct ar_hdr
+{
+    char ar_name[ 16 ];
+    char ar_date[ 12 ];
+    char ar_uid[ 6 ];
+    char ar_gid[ 6 ];
+    char ar_mode[ 8 ];
+    char ar_size[ 10 ];
+    char ar_fmag[ 2 ];
+};
+
+#define SARFMAG  2
+#define SARHDR  sizeof( struct ar_hdr )
+#endif
+
 #include <gelf.h>
 
 #include <errno.h>
diff -Naur Retro68.old/libelf/src/nlist.c Retro68/libelf/src/nlist.c
--- Retro68.old/libelf/src/nlist.c	2022-03-02 21:58:34.655141137 -0800
+++ Retro68/libelf/src/nlist.c	2022-03-02 22:22:07.587881561 -0800
@@ -37,8 +37,17 @@
 #include <nlist.h>
 #include <unistd.h>
 
+#include <sys/stat.h>
+
 #include "libelfP.h"
 
+#define _O_RDONLY      0
+#define _O_WRONLY      1
+#define _O_RDWR        2
+
+#define O_RDONLY _O_RDONLY
+#define O_WRONLY _O_WRONLY
+#define O_RDWR _O_RDWR
 
 struct hashentry
 {
@@ -71,7 +80,7 @@
   size_t cnt;
 
   /* Open the file.  */
-  fd = open (filename, O_RDONLY);
+  fd = open (filename, _O_RDONLY);
   if (fd == -1)
     {
       __libelf_seterrno (ELF_E_NOFILE);
diff -Naur Retro68.old/libelf/src/portable_endian.h Retro68/libelf/src/portable_endian.h
--- Retro68.old/libelf/src/portable_endian.h	2022-03-02 21:58:34.655141137 -0800
+++ Retro68/libelf/src/portable_endian.h	2022-03-02 22:24:42.080225345 -0800
@@ -63,6 +63,33 @@
 #	include <winsock2.h>
 #	include <sys/param.h>
 
+static long long ntohll(long long x)
+{
+	return ((unsigned long long)(((unsigned char *)(&x))[7]) << 0LL)
+	     | ((unsigned long long)(((unsigned char *)(&x))[6]) << 8LL)
+	     | ((unsigned long long)(((unsigned char *)(&x))[5]) << 16LL)
+	     | ((unsigned long long)(((unsigned char *)(&x))[4]) << 24LL)
+	     | ((unsigned long long)(((unsigned char *)(&x))[3]) << 32LL)
+	     | ((unsigned long long)(((unsigned char *)(&x))[2]) << 40LL)
+	     | ((unsigned long long)(((unsigned char *)(&x))[1]) << 48LL)
+	     | ((unsigned long long)(((unsigned char *)(&x))[0]) << 56LL);
+}
+
+static long long htonll(long long x)
+{
+	unsigned char data[8] = {};
+	data[0] = (x & 0xFF00000000000000LL) >> 56LL;
+	data[1] = (x & 0x00FF000000000000LL) >> 48LL;
+	data[2] = (x & 0x0000FF0000000000LL) >> 40LL;
+	data[3] = (x & 0x000000FF00000000LL) >> 32LL;
+	data[4] = (x & 0x00000000FF000000LL) >> 24LL;
+	data[5] = (x & 0x0000000000FF0000LL) >> 16LL;
+	data[6] = (x & 0x000000000000FF00LL) >> 8LL;
+	data[7] = (x & 0x00000000000000FFLL) >> 0LL;
+
+	return *(long long *)(data);
+}
+
 #	if BYTE_ORDER == LITTLE_ENDIAN
 
 #		define htobe16(x) htons(x)
diff -Naur Retro68.old/libelf/src/system.h Retro68/libelf/src/system.h
--- Retro68.old/libelf/src/system.h	2022-03-02 21:58:34.655141137 -0800
+++ Retro68/libelf/src/system.h	2022-03-02 22:28:31.273118160 -0800
@@ -41,6 +41,10 @@
 # define BE32(n)	be32toh (n)
 # define BE64(n)	be64toh (n)
 
+#ifndef powerof2
+#define powerof2(x)     ((((x) - 1) & (x)) == 0)
+#endif
+
 #ifndef MAX
 #define MAX(m, n) ((m) < (n) ? (n) : (m))
 #endif
@@ -70,6 +74,38 @@
      __res; })
 #endif
 
+static inline ssize_t
+pwrite (int fd, const void *buf, size_t len, off_t off)
+{
+  long long filePos = _lseeki64 (fd, off, SEEK_SET);
+  int writeResult = -1;
+  if(filePos == -1L)
+  {
+         return filePos;
+  }
+
+  writeResult = _write(fd, buf, len);
+  _lseeki64 (fd, filePos, SEEK_SET);
+
+  return writeResult;
+}
+
+static inline ssize_t
+pread (int fd, void *buf, size_t len, off_t off)
+{
+  long long filePos = _lseeki64 (fd, off, SEEK_SET);
+  int readResult = -1;
+  if(filePos == -1L)
+  {
+         return filePos;
+  }
+
+  readResult = _read(fd, buf, len);
+  _lseeki64 (fd, filePos, SEEK_SET);
+
+  return readResult;
+}
+
 static inline ssize_t __attribute__ ((unused))
 pwrite_retry (int fd, const void *buf, size_t len, off_t off)
 {
@@ -137,4 +173,8 @@
   extern int never_defined_just_used_for_checking[(expr) ? 1 : -1]	\
     __attribute__ ((unused))
 
+#define S_ISUID 0004000
+#define S_ISGID 0002000
+#define S_ISVTX 0001000
+
 #endif /* system.h */
diff -Naur Retro68.old/PEFTools/CMakeLists.txt Retro68/PEFTools/CMakeLists.txt
--- Retro68.old/PEFTools/CMakeLists.txt	2022-03-02 21:58:34.659140856 -0800
+++ Retro68/PEFTools/CMakeLists.txt	2022-03-19 14:46:31.189519315 -0700
@@ -1,8 +1,8 @@
 find_package(Boost COMPONENTS filesystem system REQUIRED)
 
-add_executable(MakePEF MakePEF.cc rs6000.h PEF.h)
+add_executable(MakePEF MakePEF.cc rs6000.h PEF.h mman.c mman.h)
 
-add_executable(MakeImport MakeImport.cc PEF.h)
+add_executable(MakeImport MakeImport.cc PEF.h mman.c mman.h)
 target_link_libraries(MakeImport ResourceFiles ${Boost_LIBRARIES})
 target_include_directories(MakeImport PRIVATE ${Boost_INCLUDE_DIR})
 
diff -Naur Retro68.old/PEFTools/MakeImport.cc Retro68/PEFTools/MakeImport.cc
--- Retro68.old/PEFTools/MakeImport.cc	2022-03-02 21:58:34.659140856 -0800
+++ Retro68/PEFTools/MakeImport.cc	2022-03-19 15:10:24.757760594 -0700
@@ -8,9 +8,11 @@
 #include <algorithm>
 
 #include <sys/types.h>
-#include <sys/mman.h>
+//#include <sys/mman.h>
+#include "mman.h"
 #include <sys/stat.h>
-#include <sys/wait.h>
+//#include <sys/wait.h>
+#include "wait.h"
 
 #include <fcntl.h>
 #include <unistd.h>
diff -Naur Retro68.old/Rez/CMakeLists.txt Retro68/Rez/CMakeLists.txt
--- Retro68.old/Rez/CMakeLists.txt	2022-03-02 21:58:34.659140856 -0800
+++ Retro68/Rez/CMakeLists.txt	2022-03-02 22:29:11.721311562 -0800
@@ -15,7 +15,7 @@
 #   You should have received a copy of the GNU General Public License
 #   along with Retro68.  If not, see <http://www.gnu.org/licenses/>.
 
-find_package(Boost COMPONENTS wave filesystem system thread regex program_options)
+find_package(Boost COMPONENTS wave filesystem system thread_win32 regex program_options)
 
 # Look for bison.
     # We need Version 3, and Mac OS X still comes with an outdated version (2.3).
diff -Naur Retro68.old/Rez/Test/UnitTests.cc Retro68/Rez/Test/UnitTests.cc
--- Retro68.old/Rez/Test/UnitTests.cc	2022-03-02 21:58:34.663140574 -0800
+++ Retro68/Rez/Test/UnitTests.cc	2022-03-19 13:50:19.565614114 -0700
@@ -1,5 +1,5 @@
 #define BOOST_TEST_MODULE UnitTests
-#define BOOST_TEST_DYN_LINK
+//#define BOOST_TEST_DYN_LINK
 #define BOOST_TEST_MAIN
 #include <boost/test/unit_test.hpp>
 
